src/parse/value/css_function.rs

```diff
@@ -0,0 +1,395 @@
+                self.toks.next();
+                return Ok(Some(buf));
+            } else if kind.is_whitespace() {
+                peek_whitespace(self.toks);
+                let next = match self.toks.peek() {
+                    Some(v) => v,
+                    None => break,
+                };
+                if next.kind == ')' {
+                    buf.push(')');
+                    self.toks.truncate_iterator_to_cursor();
+                    self.toks.next();
+                    return Ok(Some(buf));
+                } else {
+                    break;
+                }
+            } else {
+                break;
+            }
+        }
+        self.toks.reset_cursor();
+        Ok(None)
+    }
+
+    pub(super) fn try_parse_min_max(
+        &mut self,
+        fn_name: &str,
+        allow_comma: bool,
+    ) -> SassResult<Option<String>> {
+        let mut buf = if allow_comma {
+            format!("{}(", fn_name)
+        } else {
+            String::new()
+        };
+        peek_whitespace(self.toks);
+        while let Some(tok) = self.toks.peek() {
+            let kind = tok.kind;
+            match kind {
+                '+' | '-' | '0'..='9' => {
+                    self.toks.advance_cursor();
+                    if let Some(number) = self.peek_number()? {
+                        buf.push(kind);
+                        buf.push_str(&number);
+                    } else {
+                        return Ok(None);
+                    }
+                }
+                '#' => {
+                    self.toks.advance_cursor();
+                    if let Some(Token { kind: '{', .. }) = self.toks.peek() {
+                        self.toks.advance_cursor();
+                        let interpolation = self.peek_interpolation()?;
+                        match interpolation.node {
+                            Value::String(ref s, ..) => buf.push_str(s),
+                            v => buf.push_str(v.to_css_string(interpolation.span)?.borrow()),
+                        };
+                    } else {
+                        return Ok(None);
+                    }
+                }
+                'c' | 'C' => {
+                    if let Some(name) = self.try_parse_min_max_function("calc")? {
+                        buf.push_str(&name);
+                    } else {
+                        return Ok(None);
+                    }
+                }
+                'e' | 'E' => {
+                    if let Some(name) = self.try_parse_min_max_function("env")? {
+                        buf.push_str(&name);
+                    } else {
+                        return Ok(None);
+                    }
+                }
+                'v' | 'V' => {
+                    if let Some(name) = self.try_parse_min_max_function("var")? {
+                        buf.push_str(&name);
+                    } else {
+                        return Ok(None);
+                    }
+                }
```

Comment:
This logic looks quite duplicated. As well, is it necessary to parse the same character twice? Once with `v` and another time with `var`.

Reply:
Re-parsing a single character has very little overhead for this very uncommon case, and I think it makes it much more readable to use the full function name. 

I agree the code is duplicated, but the only solution I can come up with is
```rust
'e' | 'E' | 'c' | 'C' | 'v' | 'V' => {
  let fn_name = match kind {
    'c' | 'C' => "calc",
    'e' | 'E' => "env",
    'v' | 'V' => "var",
    _ => unreachable!(), // presumably, we could use `unsafe { std::hint::unreachable_unchecked!() }` here as well
  }
  ...
}
```

Reply:
This sounds like what `nom` and other parser combinators libraries solves, they do something like `parse("calc").or(parse("env")).